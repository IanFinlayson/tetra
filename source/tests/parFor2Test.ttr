def main():
   sum = 0
   print("In main(), just set sum = 0 and sum = ", sum, "\n")
   parallel for i in [[1 ... 9],[1 ... 9],[1 ... 9],[1 ... 9],[1 ... 9]]:
      parallel for j in i:
         parallel:
            parFor([1.1,2.2,3.3])
            lock sumLock:
               sum += j
               print("Holding sumLock, just added j to sum\n")

   print("Out of loop and sum = ", sum, "\n")



def parFor(x [real]):
   sum = 0.0
   parallel for d in x:
      sum += d
