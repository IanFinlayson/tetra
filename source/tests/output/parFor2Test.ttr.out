In main(), just set sum = 0 and sum = 0
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Holding sumLock, just added j to sum
Out of loop and sum = 225

